Source/Core/InputCommon/ControllerInterface/SteamDeck/SteamDeck.cpp

```diff
@@ -0,0 +1,304 @@
+  void PopulateDevices() override;
+};
+
+std::unique_ptr<ciface::InputBackend> CreateInputBackend(ControllerInterface* controller_interface)
+{
+  return std::make_unique<InputBackend>(controller_interface);
+}
+
+InputBackend::InputBackend(ControllerInterface* controller_interface)
+    : ciface::InputBackend(controller_interface)
+{
+}
+
+void InputBackend::PopulateDevices()
+{
+  INFO_LOG_FMT(CONTROLLERINTERFACE, "SteamDeck PopulateDevices");
+
+  auto possible_devices = hid_enumerate(0x28de, 0x1205);
+  bool was_found = false;
+  std::string found_path;
+  auto this_device = possible_devices;
+
+  while (this_device)
+  {
+    INFO_LOG_FMT(CONTROLLERINTERFACE, "Found {} (interface {})", this_device->path,
+                 this_device->interface_number);
+
+    if (this_device->interface_number == 2)
+    {
+      was_found = true;
+      found_path = this_device->path;
+      break;
+    }
+
+    this_device = this_device->next;
+  }
+
+  hid_free_enumeration(possible_devices);
+
+  if (was_found)
+  {
+    INFO_LOG_FMT(CONTROLLERINTERFACE, "Found Steam Deck controls at {}", found_path);
+  }
+  else
+  {
+    INFO_LOG_FMT(CONTROLLERINTERFACE, "No Steam Deck interface found");
+    return;
+  }
+
+  auto deck_dev = hid_open_path(found_path.c_str());
+  if (!deck_dev)
+  {
+    ERROR_LOG_FMT(CONTROLLERINTERFACE, "Steam Deck controls could not be opened");
+    return;
+  }
+  if (hid_set_nonblocking(deck_dev, 1) != 0)
+  {
+    ERROR_LOG_FMT(CONTROLLERINTERFACE, "Steam Deck controls could not be set nonblocking");
+    return;
+  }
+
+  GetControllerInterface().RemoveDevice(
+      [](const auto* dev) { return dev->GetSource() == STEAMDECK_SOURCE_NAME; });
+
+  GetControllerInterface().AddDevice(std::make_shared<Device>(deck_dev));
+}
+
+Device::Device(hid_device* device) : m_device{device}
+{
+  // clang-format off
+  AddInput(new Button("R2 Full Pull",       m_latest_input.buttons0, 0x00000001));
+  AddInput(new Button("L2 Full Pull",       m_latest_input.buttons0, 0x00000002));
+  AddInput(new Button("R1",                 m_latest_input.buttons0, 0x00000004));
+  AddInput(new Button("L1",                 m_latest_input.buttons0, 0x00000008));
+  AddInput(new Button("Y",                  m_latest_input.buttons0, 0x00000010));
+  AddInput(new Button("B",                  m_latest_input.buttons0, 0x00000020));
+  AddInput(new Button("X",                  m_latest_input.buttons0, 0x00000040));
+  AddInput(new Button("A",                  m_latest_input.buttons0, 0x00000080));
+  AddInput(new Button("D-Pad Up",           m_latest_input.buttons0, 0x00000100));
+  AddInput(new Button("D-Pad Right",        m_latest_input.buttons0, 0x00000200));
+  AddInput(new Button("D-Pad Left",         m_latest_input.buttons0, 0x00000400));
+  AddInput(new Button("D-Pad Down",         m_latest_input.buttons0, 0x00000800));
+  AddInput(new Button("VIEW",               m_latest_input.buttons0, 0x00001000));
```

Comment:
Does it make sense for "VIEW", "STEAM", "OPTIONS", and "DOTS" to be all caps? Maybe it does, I'm just asking.

Reply:
Do you have an official source for this? I can find a few references to Quick Access (or QAM) but none directly from Valve documentation or similar.

Reply:
Counterpoint about the Steam/Guide button:

![20230510010606_1](https://github.com/dolphin-emu/dolphin/assets/4522237/e265438b-d9a3-41e5-9abb-900ed20d8a8a)


Also I assume you're referring to this initial tutorial popup thing: https://youtu.be/GF9g_bmi-cE?t=514

Which, yes, does definitely refer to it as the 'Quick Access button'... and also definitely refers to the other button as 'Steam button'.

So I guess it's 'Steam' and 'Quick Access'?

Reply:
I would also change 'Pad' to 'Trackpad' and 'Pad Force' to 'Trackpad Pressure' based on how they're referred to in the Steam Deck UI.